syft/frameworks/torch/tensors/interpreters/precision.py

```diff
@@ -168,7 +175,10 @@ def matmul(self, *args, **kwargs):
             self.precision_fractional == other.precision_fractional
         ), "In matmul, all args should have the same precision_fractional"
 
-        return response.truncate(other.precision_fractional)
+        response = response.truncate(other.precision_fractional)
+        response %= self.field  # Wrap around the field
```

Comment:
Theoretically we want to do this wrapping automatically. I am curious if it makes sense to add this automatically to every method (this is good for now)

Reply:
Do you think I should put this somewhere in the method hooking process?

Reply:
I'm not sure it would be easy with the current functionality we have :/

Reply:
I will open an issue for this and we can address it later

Reply:
Yes looks like a good idea, basically it would be an extension of overload_torch.py I believe to execute a class specific transformation at the `def hook_args(`